vault/src/connection_manger.rs

```diff
@@ -138,18 +134,17 @@ impl<Config: Clone + Send + 'static, F: Fn()> ConnectionManager<Config, F> {
                 Box::new(constructor),
                 self.db_path.clone(),
             );
-            match service.start().await {
-                Err(err @ Error::Abort(_)) => {
-                    tracing::warn!("Disconnected: {}", err);
-                    return Err(err);
-                }
-                Err(err) => {
-                    tracing::warn!("Disconnected: {}", err);
-                }
-                _ => {
-                    tracing::warn!("Disconnected");
+
+            let backoff = ExponentialBackoff::default();
+
+            backoff::future::retry(backoff, || async {
```

Comment:
I don't think we should be using backoff here, this whole method is anyway wrapped in a loop with failure handling below based on the `RestartPolicy`.

Reply:
so should I just break the loop and return in case of `backoff::Error::Permanent` and other errors will be logged, OR  create a new Error type `ServiceError` which will be used by the `Service` trait in which `Permanent` error will be returned. 

```Rust
pub enum ServiceError {
      Permanent<crate::Error>,
      Transient<crate::Error>
}
```

Reply:
Don't break the loop since we have shutdown handling below. How did we handle the `Abort` and `Retry` variants before?

Reply:
In case of abort returning an error [here](https://github.com/interlay/interbtc-clients/blob/eeb2fda6d584dbf851ad851dc013e3cf52463de4/service/src/lib.rs#L144)

Reply:
So I think the idea was just to replace `Abort` with `Permanent` and `Retry` with `Transient` and then keep our logic the same.